### PR TITLE
asd

### PR DESCRIPTION
**Pull request by jdifek**
- [DEMO LINK](https://jdifek.github.io/js_2048_game/)

### CODE REVIEW

**Review by GUSILLUS — CHANGES_REQUESTED** [review]
GJ!

Try to implement game over message, because i don't have available moves but I still can't play

**Comment by TarasHoliuk** [comment]
1. When I lose the game and press `Restart`, then start clicking arrow buttons I see this (button still says Start and I see the message below), but I can play:

image

2. You shouldn't add a new tile when after the arrow button click nothing changed:
- Current field:
image

- I pressed left (nothing changed, but the new tile was added):
image

File: src/modules/Game.class.js [hunk 1]
```diff
@@ -1,68 +1,270 @@
...
+
+    this.addRandomTile();
+
+    if (!this.movesAvailable()) {
+      this.status = Game.STATUS_LOSE;
+    }
+  }
+
+  moveDown() {
+    if (this.status === Game.STATUS_WIN || this.status === Game.STATUS_LOSE) {
```
Comment (TarasHoliuk) [inline src/modules/Game.class.js:149]: Maybe you shouldn't do anything when the status is IDLE as well. See the error in the comments

File: src/scripts/main.js [hunk 2]
```diff
@@ -1,7 +1,84 @@
...
+  } else {
+    button.textContent = 'Start';
+
+    game.restart();
+    updateGame();
+  }
+});
+
+function handleKeyPress(e) {
+  if (e.key === 'ArrowDown') {
```
Comment (TarasHoliuk) [inline src/scripts/main.js:30]: You can use switch here

**Review by TarasHoliuk — CHANGES_REQUESTED** [review]
For some reason, tests can't start
Try to fix it (ask for the help in the chat if you need)

**Pull request closed without merging**